### PR TITLE
Bind many more properties to scripts

### PR DESCRIPTION
**Pull request by bojidar-bg**
Notable potentially breaking changes:
- PROPERTY_USAGE_NOEDITOR is now PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_NETWORK, without PROPERTY_USAGE_INTERNAL
- Some properties were renamed, and sometimes even shadowed by new ones
- New getter methods (some virtual) were added

Some notes I did while doing this:
* AnimationPlayer.playback/play became AnimationPlayer.current_animation, with a backward-compatibility fallback. I think I tested this part, but slight bugs are possible.
* Node.filename, Node.name and Node.owner might generate some GDScript errors about shadowing.
* ScrollContainer.h_scroll and ScrollContainer.v_scroll are now scroll_horizontal_enabled and scroll_vertical_enabled.
* CanvasLayer.rotation and Polygon2D.texture_rotation were in degrees, while the new property is in radians. Likely to break a few projects.
* I couldn't really test Control.default_cursor_shape on linux.
* Used OS.window_borderless instead of OS.borderless_window, so we might add a group at a later point.
* Unbound TextEdit.set_max_chars, but added potential bindings for later on, since it wasn't used anywhere. Now I see there is an issue for set_wrap as well, might unbind it as well...
* OS.window_title, Physics{,2D}ShapeQueryParameters.shape and Viewport.attach_to_screen_rect lack getters, so I didn't bind them.
* ~~Added a setter for VehicleBody.linear_velocity, hopefully it works.~~ All the while someone made it extend RigidBody.

While reviewing, please pay attention to the usage of the properties. It should not be `PROPERTY_USAGE_NOEDITOR` or `DEFAULT` for properties that are not supposed to be saved to disk.

### CODE REVIEW

**Comment by akien-mga** [comment]
Let's test in production™ and see what breaks :) This is quite important for 3.0 to have a consistent API.

File: doc/classes/@GlobalScope.xml [hunk 1]
```diff
@@ -29,9 +29,6 @@
 		<member name="Geometry" type="Geometry" setter="" getter="">
 			[Geometry] singleton
 		</member>
-		<member name="GodotSharp" type="GodotSharp" setter="" getter="">
-			[GodotSharp] singleton
```
Comment (akien-mga) [inline doc/classes/@GlobalScope.xml]: Hm..

File: doc/classes/SceneTree.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (akien-mga) [inline doc/classes/SceneTree.xml]: Hm.. (bis)

File: doc/classes/@GlobalScope.xml (hunk 1 above)
Comment (bojidar-bg) [reply doc/classes/@GlobalScope.xml]: Not sure how those happened, I just ran doctool a few times.

Comment (akien-mga) [reply doc/classes/@GlobalScope.xml]: Only the mono build outputs this GodotSharp singleton. For the other issue, not sure.

**Pull request merged**